server/src/main/java/org/elasticsearch/action/ingest/GetPipelineAction.java

```diff
@@ -24,7 +24,7 @@
 public class GetPipelineAction extends ActionType<GetPipelineResponse> {
 
     public static final GetPipelineAction INSTANCE = new GetPipelineAction();
-    public static final String NAME = "cluster:admin/ingest/pipeline/get";
+    public static final String NAME = "cluster:monitor/ingest/pipeline/get";
```

Comment:
This particular rename won't work flawlessly in a mixed cluster upgrade scenario, because the action is a `TransportMasterNodeReadAction`.
Say a non-master node has been upgraded, but the master hasn't yet. If the get pipelines REST request hits the upgraded non-master node, it will forward the transport request to the master node, but it will ask for the renamed action of which master doesn't know about.

Reply:
@albertzaharovits That's what I've just figured out. How do we normally handle such changes?